src/components/views/elements/PersistentApp.tsx

```diff
@@ -71,6 +83,17 @@ export default class PersistentApp extends React.Component<IProps, IState> {
         });
     };
 
+    private onWidgetAction = (payload: ActionPayload): void => {
+        switch (payload.action) {
+            case Action.AfterRightPanelPhaseChange:
```

Comment:
sadly this will need some refactoring now when merging with the right panel refactor...